pyramda/iterable/flatten.py

```diff
@@ -0,0 +1,26 @@
+from collections import Iterable
+from past.builtins import basestring
+
+from pyramda.function.curry import curry
+
+
+@curry
+def flatten(xs, depth):
```

Comment:
What about instead of a depth, we name this `flatten_until` and pass a predicate that determines whether a value is a leaf? Presuming a curried version of isinstance was provided, we could implement flattening that handled strings like this:

```
flatten = flatten_until(isinstance(basestring))
```

The `flatten_until` implementation would probably look something like this:

```
def flatten_until(is_leaf, v):
    if isinstance(v, Iterable) and not is_leaf(v):
      ... recursively call flatten_until on elements of v and yield them all ...
    else:
      yield v
```

What do you think?